src/code/views/node-view.coffee

```diff
@@ -16,6 +16,11 @@ NodeTitle = React.createFactory React.createClass
     if @props.isEditing
       @inputElm().off()
 
+  componentWillUpdate: (nextProps) ->
+    # mark the title as updated even if no change was made when it leaves edit mode
+    @props.node.titleUpdated = true if @props.isEditing and not nextProps.isEditing
+    @canDeleteWhenEmpty = @props.node.addedThisSession and not @props.node.titleUpdated
```

Comment:
I feel like the `@canDeleteWhenEmpty` variable adds just a little bit of extra confusion to what's otherwise a simple check, and it takes an extra moment of thinking to see how it relates to a handler being added or not on every re-render of the view. It seems to me that it would just be a little simpler to simply have the check for both  `addedThisSession ` and `titleUpdated` in `detectDeleteWhenEmpty` (or maybe `detectDeleteIfNew`). 

For instance, you're already checking `titleUpdated` again in that method, which I *think* is now unnecessary, since the handler will be removed on the first rerender after the title is updated, but I feel like it shouldn't be necessary to work out whether the handler will always be removed at that point or not. 

So, simpler would just be to do both checks in the handler, and always have the hander attached.

But if you think this makes more sense to you, I don't feel too strongly about it.